ignite/services/chain/init.go

```diff
@@ -19,35 +19,33 @@ const (
 	moniker = "mynode"
 )
 
-// Init initializes the chain and applies all optional configurations.
-func (c *Chain) Init(ctx context.Context, initAccounts bool) error {
-	conf, err := c.Config()
-	if err != nil {
-		return &CannotBuildAppError{err}
-	}
-
-	if err := c.InitChain(ctx); err != nil {
+// Init initializes the chain
+// initAccounts initializes chain accounts from the Ignite config
+// initConfiguration initializes node configuration from the Ignite config
+// initGenesis initializes genesis state for the chain from Ignite config
+func (c *Chain) Init(ctx context.Context, initAccounts, initConfiguration, initGenesis bool) error {
```

Comment:
@lubtd Can we consider using a struct instead of 3 bools ?

Something like : 

```go
type InitArgs struct {
  InitAccounts bool
  InitConfiguration bool
  InitGenesis bool
}
```

Note I call it `Args` to distinct with `Options` that usually are functions.

For the callers that makes the code a lot clearer (but more verbose I admit) : 

```diff
- c.Init(cmd.Context(), true, true, true)
+ c.Init(cmd.Context(), InitArgs{ InitAccounts: true,  InitConfiguration: true, InitGenesis: true })
```

And at least when everything is false you can omit the fields

```diff
- c.Init(cmd.Context(), false, false, false)
+ c.Init(cmd.Context(), InitArgs{})
```

WDYT?

Reply:
I would like to add that there are other places where a pattern like this would help cleaning up arguments where variadic options might not be the best fit.

In any case I think variadic options should be used instead of a struct but it's true that some times option names might clash so in those cases using `XXArgs{}` sounds like a good choice.

Reply:
Yeah, variadic options is a bit too much for some case as we must define option setter method for each option